app/integration/project-generator/src/main/java/io/syndesis/integration/project/generator/ProjectGenerator.java

```diff
@@ -0,0 +1,249 @@
+            resource = getClass().getResource("templates/" + overridePath + "/" + template);
+        }
+        if (resource == null) {
+            resource = getClass().getResource("templates/" + template);
+        }
+        if (resource == null) {
+            throw new IllegalArgumentException(
+                String.format("Unable to find te required template (overridePath=%s, template=%s)"
+                    , overridePath
+                    , template
+                )
+            );
+        }
+
+        try (InputStream stream = resource.openStream()) {
+            return mf.compile(new InputStreamReader(stream, StandardCharsets.UTF_8), name);
+        }
+    }
+
+    private void addAdditionalResources(TarArchiveOutputStream tos) throws IOException {
+        for (ProjectGeneratorConfiguration.Templates.Resource additionalResource : generatorProperties.getTemplates().getAdditionalResources()) {
+            String overridePath = generatorProperties.getTemplates().getOverridePath();
+            URL resource = null;
+
+            if (!StringUtils.isEmpty(overridePath)) {
+                resource = getClass().getResource("templates/" + overridePath + "/" + additionalResource.getSource());
+            }
+            if (resource == null) {
+                resource = getClass().getResource("templates/" + additionalResource.getSource());
+            }
+            if (resource == null) {
+                throw new IllegalArgumentException(
+                    String.format("Unable to find te required additional resource (overridePath=%s, source=%s)"
+                        , overridePath
+                        , additionalResource.getSource()
+                    )
+                );
+            }
+
+            try {
+                ProjectGeneratorHelper.addTarEntry(tos, additionalResource.getDestination(), Files.readAllBytes(Paths.get(resource.toURI())));
+            } catch (URISyntaxException e) {
+                throw new IOException(e);
+            }
+        }
+    }
+
+    @SuppressWarnings("PMD.DoNotUseThreads")
+    private Runnable generateAddProjectTarEntries(IntegrationDeployment deployment, OutputStream os) {
+        return () -> {
+            try (
+                TarArchiveOutputStream tos = new TarArchiveOutputStream(os)) {
+                tos.setLongFileMode(TarArchiveOutputStream.LONGFILE_POSIX);
+
+                ProjectGeneratorHelper.addTarEntry(tos, "src/main/java/io/syndesis/example/Application.java", ProjectGeneratorHelper.generate(deployment, applicationJavaMustache));
+                ProjectGeneratorHelper.addTarEntry(tos, "src/main/resources/application.properties", ProjectGeneratorHelper.generate(deployment, applicationPropertiesMustache));
+                ProjectGeneratorHelper.addTarEntry(tos, "src/main/resources/syndesis/integration/deployment.json", Json.mapper().writerWithDefaultPrettyPrinter().writeValueAsBytes(deployment));
```

Comment:
is the deployment.json for reference only or how it is used ? Does is has the same format as the `deployment.json` from rest ? (if not, maybe its better to rename it to avoid confusion)

Reply:
no it has different format, will rename it

Reply:
done